src/org/parosproxy/paros/view/FindDialog.java

```diff
@@ -108,6 +124,48 @@ private void initialize() {
         txtFind.requestFocus();
         this.getRootPane().setDefaultButton(btnFind);
         pack();
+        this.setVisible(true);
+	}
+	
+	private void registerParent(Object parent) {
+		if (!getParentsList().containsKey(parent)) {
+			getParentsList().put(parent, this);
+		}
+	}
+	
+	private static Map<Object, FindDialog> getParentsList() {
+		return parentsList;
+	}
+	
+	/**
+	 * Get the FindDialog for the parent if there is one or creates and returns a new one.
+	 * 
+	 * @param parent the parent Window (or Frame) for this FindDialog
+	 * @param modal a boolean indicating whether the FindDialog should ({@code true}), 
+	 * or shouldn't ({@code false}) be modal.
+	 * @return The existing FindDialog for the parent (if there is one), or a new FindDialog.
+	 * @since TODO add version
+	 */
+	public static FindDialog getDialog(Window parent, boolean modal) {
+		if (getParentsList().containsKey(parent)) {
+			FindDialog activeDialog = getParentsList().get(parent);

```

Comment:
Might worth moving this statement to before the if (and check that's non-`null` in the if statement), that ensures the object does not get garbage collected in the meantime.